src/Fmt_ast.ml

```diff
@@ -1603,16 +1603,34 @@ and fmt_expression c ?(box= true) ?epi ?eol ?parens ?ext ({ast= exp} as xexp)
         $ fmt_atrs )
   | Pexp_open (flag, {txt; loc}, e0) ->
       let override = Poly.(flag = Override) in
+      (* open_style = `Long means we use let open Module in (...) style;
+         open_style = `Short means we use Module.(...) style; open_style =
+         `Auto means we use the one fitting best. *)
+      let open_style =
+        match c.conf.open_style with
+        | `Auto -> `Auto
+        | `Short -> `Short
+        | `Preserve ->
+            let filter = function Parser.IN -> true | _ -> false in
+            let loc = {loc with loc_end= e0.pexp_loc.loc_start} in
+            let toks = Source.tokens_at c.source loc ~filter in
+            if List.is_empty toks then `Short else `Long
```

Comment:
It would be nice if these 4 lines could be outlined into a function in `Source` with a descriptive name.

Reply:
I added the functions `Source.loc_between` and `Source.tokens_between`, it didn't shorten the code but I did not want to be too specific here.